integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/DatabaseTest.java

```diff
@@ -60,72 +60,98 @@ public void tear(Object[] data) {
+            then = "the database should be created again")
+    public void createAndDeleteAndCreateWithSameNameThenShouldRecreatedDatabase(TestContext testContext) {
         String databaseName = getNameGenerator().getRandomNameForResource();
         testContext
                 .given(DatabaseEntity.class)
                 .withName(databaseName)
-                .when(DatabaseEntity.post())
-                .when(DatabaseEntity.list())
-                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1))
-                .when(DatabaseEntity.deleteV2())
-                .when(DatabaseEntity.list())
-                .then(DatabaseExistsAssertion.getAssertion(databaseName, 0))
-                .when(DatabaseEntity.post())
-                .when(DatabaseEntity.list())
-                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1))
+                .when(DatabaseEntity.post(), key(databaseName))
+                .when(DatabaseEntity.list(), key(databaseName))
+                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1), key(databaseName))
+                .when(DatabaseEntity.deleteV2(), key(databaseName))
+                .when(DatabaseEntity.list(), key(databaseName))
+                .then(DatabaseExistsAssertion.getAssertion(databaseName, 0), key(databaseName))
+                .when(DatabaseEntity.post(), key(databaseName))
+                .when(DatabaseEntity.list(), key(databaseName))
+                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1), key(databaseName))
                 .validate();
     }
 
     @Test(dataProvider = TEST_CONTEXT_WITH_MOCK)
-    public void testCreateTwice(TestContext testContext) {
+    @Description(
+            given = "there is a prepared database",
+            when = "when a database create request is sent with the same database name",
+            then = "the create should return a BadRequestException")
+    public void createAndCreateWithSameNameThenShouldThrowBadRequestException(TestContext testContext) {
         String databaseName = getNameGenerator().getRandomNameForResource();
         testContext
                 .given(DatabaseEntity.class)
                 .withName(databaseName)
-                .when(DatabaseEntity.post())
-                .when(DatabaseEntity.list())
-                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1))
-                .when(DatabaseEntity.post(), key(BAD_REQUEST_KEY))
-                .expect(BadRequestException.class, key(BAD_REQUEST_KEY))
+                .when(DatabaseEntity.post(), key(databaseName))
+                .when(DatabaseEntity.list(), key(databaseName))
+                .then(DatabaseExistsAssertion.getAssertion(databaseName, 1), key(databaseName))
+                .when(DatabaseEntity.post(), key(databaseName))
+                .expect(BadRequestException.class, key(databaseName))
```

Comment:
if the first `post` is throwing an exception then the test succeeds. You should not use the same keys multiple times.